Don't define OpenCL keywords before including system headers

In file included from unit_tests/built_ins/built_in_kernels_tests.cpp:10:
In file included from third_party/gtest/gtest/gtest.h:57:
In file included from /usr/include/c++/v1/ostream:138:
In file included from /usr/include/c++/v1/ios:215:
/usr/include/c++/v1/__locale:177:29: error: expected member name or ';' after declaration specifiers
    static locale& __global();
    ~~~~~~~~~~~~~           ^
In file included from unit_tests/built_ins/built_in_kernels_tests.cpp:10:
In file included from third_party/gtest/gtest/gtest.h:57:
In file included from /usr/include/c++/v1/ostream:140:
In file included from /usr/include/c++/v1/locale:197:
In file included from /usr/include/nl_types.h:40:
In file included from /usr/include/sys/types.h:377:
/usr/include/sys/select.h:105:37: error: too many arguments provided to function-like macro invocation
int     select(int, fd_set *, fd_set *, fd_set *, struct timeval *);
                                        ^
runtime/builtin_kernels_simulation/opencl_c.h:223:9: note:
      macro 'select' defined here
#define select(a, b, c) (c ? b : a)
        ^

--- unit_tests/built_ins/built_in_kernels_tests.cpp.orig	2019-11-29 14:23:34 UTC
+++ unit_tests/built_ins/built_in_kernels_tests.cpp
@@ -5,9 +5,9 @@
  *
  */
 
-#include "runtime/builtin_kernels_simulation/opencl_c.h"
-
 #include "gtest/gtest.h"
+
+#include "runtime/builtin_kernels_simulation/opencl_c.h"
 
 namespace BuiltinKernelsSimulation {
 
